### PR TITLE
Collaborative Editor: create workflow

### LINKED ISSUE #3688: Refactor YAML parsing utilities for React integration
### User story
As a developer, I want to reuse the existing YAML parsing and validation TypeScript utilities in the React implementation so that we maintain consistent behavior and avoid code duplication.

### Details
The current implementation has comprehensive TypeScript utilities for YAML parsing, schema validation, error handling, and state conversion. These utilities should be reused in the React implementation with minimal or no changes, as they are pure functions independent of the UI framework.

**Existing Utilities (Can Be Reused):**
- `assets/js/yaml/util.ts`:
  - `parseWorkflowYAML()` - YAML parsing + validation
  - `convertWorkflowSpecToState()` - Spec to state conversion
  - `convertWorkflowStateToSpec()` - Reverse conversion
  - `parseWorkflowTemplate()` - Template parsing (no schema validation)
- `assets/js/yaml/workflow-errors.ts`:
  - WorkflowError class hierarchy
  - Error formatting functions
  - Error type enums
- `assets/js/yaml/types.ts`:
  - TypeScript type definitions
  - WorkflowSpec, WorkflowState interfaces
- `assets/js/yaml/schema/workflow-spec.json`:
  - JSON Schema for validation

**Phoenix Hook Code (Needs React Adaptation):**
- `YAMLToWorkflow.ts` - Hook-specific code needs to become React component
- Event listeners → React event handlers
- DOM manipulation → React state management
- pushEventTo → Props callbacks or LiveView pushEvent
- handleEvent → Props or context

### Implementation notes
- util.ts functions are pure and framework-agnostic - can import directly
- workflow-errors.ts is pure TypeScript - reuse as-is
- types.ts interfaces - reuse directly
- Schema file - import path may need adjustment
- YAMLToWorkflow hook logic needs to be refactored into React component
- Consider creating custom React hooks: `useYAMLValidation`, `useFileUpload`
- FileDropzone.ts logic should become React component or hook
- Maintain same error handling behavior
- Keep same validation order and error prioritization
- Test coverage should carry over (adapt tests to React)

### Release notes
Refactored YAML parsing utilities for React integration while maintaining compatibility

### User acceptance criteria
- [ ] parseWorkflowYAML() function works identically in React
- [ ] convertWorkflowSpecToState() produces same output
- [ ] All error classes throw correctly in React context
- [ ] Schema validation behavior unchanged
- [ ] Type definitions compile correctly
- [ ] No duplicate code between hook and React versions
- [ ] Validation error messages match exactly
- [ ] File reading behavior identical
- [ ] All existing edge cases handled (empty names, missing IDs, etc.)
- [ ] Import paths resolve correctly in React build
- [ ] Tree-shaking works (unused code not bundled)
- [ ] Tests adapted and passing for React implementation

### LINKED ISSUE #3697: Implement automatic workflow name uniqueness handling
### User story
As a user, I want my workflow name to be automatically made unique if it conflicts with existing workflows so that I can successfully create workflows without manual name adjustments.

### Details
When YAML content is successfully parsed and contains a workflow name that already exists in the project, the system automatically appends a number (starting with 1) to make the name unique. For example, "My Workflow" becomes "My Workflow 1" if the original exists. If the YAML has no name or an empty name, it defaults to "Untitled workflow".

**Current Implementation:**
- **Elixir (Server-side)**: `ensure_unique_name/2` in `new_workflow_component.ex:196-228`
- Runs during `workflow-parsed` event handling (line 115)
- Trims and validates name from params
- Queries all existing workflows in project
- Generates sequential numbered names: "Name 1", "Name 2", etc.
- Default fallback: "Untitled workflow"

**React Components Needed:**
- None - this is server-side logic, but React component should handle the name update in the response

### Implementation notes
- Current implementation happens on server after client parsing succeeds
- Name uniqueness check requires database query (must stay server-side)
- Sequential number generation: Stream.iterate starting at 1 (line 230-235)
- Name trimming and empty string handling (line 198-204)
- The modified name is returned in the changeset and should update client state
- Need to ensure React state syncs with server-assigned unique name
- This occurs before Ecto changeset validation (line 116)

### Release notes
Added automatic workflow name uniqueness handling with sequential numbering

### User acceptance criteria
- [ ] Duplicate workflow names automatically appended with " 1"
- [ ] Sequential numbering if multiple duplicates exist
- [ ] Empty or whitespace-only names default to "Untitled workflow"
- [ ] Name changes communicated back to client
- [ ] Client state updated with server-assigned unique name
- [ ] User can see final name before clicking Create
- [ ] Numbering continues from highest existing number (e.g., if "Name 1" and "Name 3" exist, next is "Name 4")

Depends on #3695

### LINKED ISSUE #3700: Implement workflow creation from validated YAML
### User story
As a user, I want to successfully create a workflow when clicking Create with valid YAML so that my workflow is saved and I can begin editing it.

### Details
When the Create button is clicked with valid YAML content, the system creates a new workflow in the database, redirects to the workflow editor, and shows a success message. The workflow is created with all jobs, triggers, and edges as defined in the YAML.

**Current Implementation:**
- **LiveView (Parent edit.ex)**: Save handler at `edit.ex:1910-1956`
- Button triggers: `JS.push("save")` event to parent LiveView (not NewWorkflowComponent)
- Save flow:
  1. Button push → `handle_event("save")` in edit.ex
  2. → `save_workflow/2` (edit.ex:2461-2503)
  3. → `Helpers.save_workflow/2` (actual DB persistence)
  4. → Redirect to edit mode with new workflow ID
- Workflow params: Applied via `WorkflowParams.apply_form_params/2` (workflow_params.ex:12-25)
- Changeset: Built and validated before save (edit.ex:2379-2400)
- Success: Flash message "Workflow saved successfully."
- Post-save: URL changes from `/w/new` to `/w/:id`, live_action changes to `:edit`

**React Components Needed:**
- Create button (covered in previous issue)
- May need to trigger LiveView events or callbacks

### Implementation notes
- Save event handled by parent LiveView, not NewWorkflowComponent
- Authorization check: `check_user_can_save_workflow/1` before saving
- Workflow params merged with submitted params using deep merge
- Changeset built from merged params + project_id
- Default adaptors set if not specified (via `set_default_adaptors/0`)
- On success: assigns updated, page_title set, maybe_push_workflow_created
- Error cases handled: changeset errors, deleted workflow, authorization failures
- Need to determine how React component communicates with parent LiveView
- Consider using pushEvent or callback props pattern

### Release notes
Added workflow creation from YAML with validation and database persistence

### User acceptance criteria
- [ ] Clicking Create with valid YAML saves workflow to database
- [ ] New workflow includes all jobs from YAML
- [ ] New workflow includes all triggers from YAML
- [ ] New workflow includes all edges from YAML
- [ ] Workflow positions preserved if included in YAML
- [ ] Default adaptors applied if not specified
- [ ] Success message displayed after creation
- [ ] Page redirects to workflow editor (edit mode)
- [ ] URL updates to include new workflow ID
- [ ] Authorization checked before save
- [ ] Graceful error handling for save failures
- [ ] Optimistic locking prevents concurrent edit conflicts

Depends on #3698, #3697

### LINKED ISSUE #3738: YAML Workflow Creation: Workflow state reset when clicking settings panel after upload
## User story

As a user creating a new workflow from a YAML file, I expect the workflow state to persist when I open the workflow settings panel, so that my workflow configuration doesn't get wiped out.

## Current behavior

After uploading a YAML file to create a new workflow:
1. The workflow is loaded into the collaborative editor
2. When clicking on the workflow settings panel link/button, the workflow state is wiped out
3. The editor becomes empty or reverts to initial state

## Problem

The settings panel assumes the workflow is in "edit" mode (existing workflow), but after YAML upload we're still in "new" mode (creating a workflow). This mode mismatch causes the workflow state to be cleared.

## Expected behavior

- Clicking the workflow settings panel should preserve all workflow state
- Settings panel should work correctly in both "new" and "edit" modes
- User should be able to modify workflow settings before saving the newly created workflow

## Technical notes

- Related to workflow live_action mode handling ( vs )
- May need to handle settings panel initialization differently for new workflows
- Workflow state should be maintained in Y.Doc regardless of panel interactions

## Acceptance criteria

- [ ] After YAML upload, clicking workflow settings preserves workflow state
- [ ] Settings panel correctly identifies and handles "new" workflow mode
- [ ] All workflow data (jobs, triggers, edges) remains intact
- [ ] Settings can be modified before initial save
- [ ] No console errors or state synchronization issues

### LINKED ISSUE #3739: YAML Workflow Creation: Cannot edit workflow after creation - Save button shows 'outdated snapshot'
## User story

As a user who has just created a workflow from a YAML file, I expect to be able to continue editing the workflow after saving it, so that I can make additional changes without being blocked.

## Current behavior

After creating and saving a new workflow from YAML upload:
1. The workflow is successfully saved to the database
2. The save button becomes disabled
3. The save button shows tooltip: "outdated snapshot" or similar message
4. User cannot make further edits to the workflow

## Problem

The save operation successfully creates the workflow but doesn't properly transition the collaborative editor session from "new" workflow mode to "edit" workflow mode. This leaves the editor in an inconsistent state where:
- The workflow exists in the database
- The editor still thinks it's working with a "new" workflow
- Snapshot version tracking is out of sync
- The save button remains disabled

## Expected behavior

- After saving a new workflow, the editor should automatically transition to "edit" mode
- The snapshot version should be updated to reflect the current saved state
- The save button should become enabled again when new changes are made
- User can continue editing the workflow normally

## Technical notes

This is related to the WorkflowChannel and Session process handling in the Elixir backend:
- Location: lib/lightning_web/channels/workflow_channel.ex
- Session process: lib/lightning/collaboration/session.ex
- Need to handle the state transition when a workflow is saved for the first time
- Must update snapshot tracking and version information
- May need to send updated session state to the client after save

The transition from "new" to "edit" involves:
- Changing live_action from :new to :edit
- Updating the workflow lock_version
- Creating initial snapshot
- Updating session state in the collaborative editing system
- Broadcasting state changes to all connected clients

## Acceptance criteria

- [ ] After saving a new workflow, the editor transitions to edit mode
- [ ] Snapshot version is correctly initialized and tracked
- [ ] Save button re-enables when new changes are made
- [ ] No "outdated snapshot" errors after initial save
- [ ] Session state is correctly synchronized between server and client
- [ ] Multiple users can join and edit after workflow creation
- [ ] Workflow lock_version is properly managed

### PR DESCRIPTION
**Pull request by midigofrank**
## Description

Adds remaining workflow creation via YAML issues

Closes #3700 
Closes #3739
Closes #3738 
Closes #3688
Closes #3697

## Validation steps

1. In your project, click on create new workflow. 
2. In the create new workflow page, add `/collaborate` to your url. The full url should be: `http://localhost:4000/projects/4adf2644-ed4e-4f97-a24c-ab35b3cb1efa/w/new/collaborate` , but with your `project-id`
3. On the left panel, click on the `import button`
4. That should open the import workflow via yaml panel.

## AI Usage

Please disclose how you've used AI in this work (it's cool, we just want to know!):

- [x] Code generation (copilot but not intellisense)
- [x] Learning or fact checking
- [ ] Strategy / design
- [x] Optimisation / refactoring
- [ ] Translation / spellchecking / doc gen
- [ ] Other
- [ ] I have not used AI

You can read more details in our [Responsible AI Policy](https://www.openfn.org/ai#pull-request-templates)

## Pre-submission checklist

- [x] I have performed a **self-review** of my code.
- [x] I have implemented and tested all related **authorization policies**. (e.g., `:owner`, `:admin`, `:editor`, `:viewer`)
- [x] I have updated the **changelog**.
- [x] I have ticked a box in "AI usage" in this PR

### CODE REVIEW

File: assets/js/collaborative-editor/stores/createWorkflowStore.ts [hunk 1]
```diff
@@ -605,13 +605,41 @@ export const createWorkflowStore = () => {
...
+    }
+
+    const workflowMap = ydoc.getMap("workflow");
+
+    ydoc.transact(() => {
+      workflowMap.set("lock_version", lockVersion);
+    });
+
+    logger.debug("Updated workflow lock_version in Y.Doc", { lockVersion });
+  };
```
Comment (stuartc) [inline assets/js/collaborative-editor/stores/createWorkflowStore.ts]: I think setting this in the ydoc on the client is the wrong way around - the server updates the lock_version during save_workflow, so any changes from the resulting saved workflow should be merged/put into the ydoc inside the session on the server; and the lock version makes its way back to the client.

The "latest lock version" thats in the SessionContextStore, that one is set by the return result of the save. So the current workflow (or snapshot when that comes) is updated via the ydoc, and the `latestSnapshotLockVersion` property in SessionContext gets updated by a broadcast and/or the response payload from saveWorkflow (save_workflow handler).
